RealmWeaver/ModuleWeaver.cs

```diff
@@ -178,11 +178,12 @@ public void Execute()
         var listTypeDefinition = _corLib.MainModule.GetType("System.Collections.Generic.List`1");
         if (listTypeDefinition == null)
         {
-            var collectionsAssembly = AssemblyResolver.Resolve("System.Collections");
-            listTypeDefinition = collectionsAssembly.MainModule.ExportedTypes.FirstOrDefault(x => x.FullName == "System.Collections.Generic.List`1").Resolve();
-            
+            System_IList = _realmAssembly.MainModule.ImportReference(typeof (System.Collections.Generic.List<>));
```

Comment:
It is a fallback when we couldn't get the main definition so is only done in the PCL - empirically it seems to work.